kiwi/runtime_checker.py

```diff
@@ -412,6 +414,34 @@ def check_efi_mode_for_disk_overlay_correctly_setup(self):
         if overlayroot and firmware == 'uefi':
             raise KiwiRuntimeError(message)
 
+    def check_grub_efi_installed_for_efi_firmware(self):
+        """
+        If the image is being built with efi or uefi firmware setting
+        we need a grub(2)-*-efi package installed. The check is not 100%
+        as every distribution has different names and different requirement
+        but it is a reasonable approximation on the safe side meaning the
+        user may still get an error but should not receive a false positive
+        """
+        message = dedent('''\n
+            Firmware set to efi or uefi but not grub*efi* package is
+            part of the image build.
+        ''')
+        firmware = self.xml_state.build_type.get_firmware()
+        if firmware in ('efi', 'uefi'):
+            grub_efi_packages = (
+                'grub2-x86_64-efi',  # SUSE
+                'grub-efi',  # Debian based distros have grub-efi-*
+                'grub2-efi'  # RedHat
+            )
```

Comment:
I am fine with this change, I also prefer to not skip the run time check according to the package manager type. I guess those package names won't change that much over time.